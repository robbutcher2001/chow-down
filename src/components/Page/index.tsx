import React, { ReactNode } from 'react';

import styled, { ThemeProvider } from 'styled-components';
import theme from '../../theme';

import Footer from '../Footer';
import Header from '../Header';
import Nav from '../Nav';

interface PageProps {
  children: ReactNode
};

const Page = styled.div`
  display: grid;
  align-items: flex-start;
  grid-template-rows: auto auto 1fr auto;
  min-height: 100vh;
`

//test - will need an event listener for changes though
// const isDarkTheme = window.matchMedia('(prefers-color-scheme: dark)').matches;
// console.log(isDarkTheme);
// console.log(theme);

export default (props: PageProps) => (
  <Page>
    <ThemeProvider theme={theme}>
      <Nav />
      <Header />
      {props.children}
      <Footer />
    </ThemeProvider>
  </Page>
);