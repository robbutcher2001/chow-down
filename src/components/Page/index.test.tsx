import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import Page from '.';

test('Page basic snapshot render', () => {
  const page = renderer.create(
    <Router history={createMemoryHistory()}>
      <Page>
        <div>fake_child</div>
      </Page>
    </Router>
  );

  expect(page.toJSON()).toMatchSnapshot();
});

test('Page correct tag content assertion', () => {
  const { getByText } = render(
    <Router history={createMemoryHistory()}>
      <Page>
        <div>fake_child</div>
      </Page>
    </Router>
  );

  expect(getByText(/Your Week/).textContent).toEqual('Your Week');
  expect(getByText(/Recipes/).textContent).toEqual('Recipes');
  expect(getByText(/fake_child/).textContent).toEqual('fake_child');
  expect(getByText(/Chow Down/).textContent).toEqual('Chow Down');
  expect(getByText(/Chow down on/).textContent).toEqual('Chow down on a weekly plan of your evening meals.');
  expect(getByText(/Rob Butcher/).textContent).toEqual('Â© Rob Butcher 2020.');
  expect(getByText(/Images/).textContent).toEqual('Images: Unsplash and various recipe websites.');
});