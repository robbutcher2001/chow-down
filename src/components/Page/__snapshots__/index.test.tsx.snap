// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page basic snapshot render 1`] = `
<div
  className="sc-bxivhb jUPexl"
>
  <nav
    className="sc-htpNat fNawMe"
  >
    <a
      className="active"
      href="/"
      onClick={[Function]}
    >
      Your Week
    </a>
    <a
      className=""
      href="/recipes"
      onClick={[Function]}
    >
      Recipes
    </a>
    <a
      className=""
      href="/settings"
      onClick={[Function]}
    >
      <img
        src="test-file-stub"
      />
    </a>
  </nav>
  <header
    className="sc-bwzfXH bqDNJO"
  >
    <h1>
      <a
        href="/"
        onClick={[Function]}
      >
        Chow Down
      </a>
    </h1>
    <p>
      Chow down on a weekly plan of your evening meals.
    </p>
  </header>
  <div>
    fake_child
  </div>
  <footer
    className="sc-bdVaJa hZcchM"
  >
    <p>
      Â© Rob Butcher 2020.
    </p>
    <p>
      Images: 
      <a
        href="https://unsplash.com/"
        target="_blank"
      >
        Unsplash
      </a>
       and various recipe websites.
    </p>
  </footer>
</div>
`;
